app/models/sidebar_url.rb

```diff
@@ -7,14 +7,32 @@ class SidebarUrl < ActiveRecord::Base
 
   validate :path_validator
 
+  before_save :remove_internal_hostname, :set_external
+
   def path_validator
-    Rails.application.routes.recognize_path(value)
+    if external?
+      raise ActionController::RoutingError if value !~ Discourse::Utils::URI_REGEXP
+    else
+      Rails.application.routes.recognize_path(value)
+    end
   rescue ActionController::RoutingError
     errors.add(
       :value,
       I18n.t("activerecord.errors.models.sidebar_section_link.attributes.linkable_type.invalid"),
     )
   end
+
+  def remove_internal_hostname
+    self.value =
+      self
+        .value
+        .sub("http://#{Discourse.current_hostname}", "")
+        .sub("https://#{Discourse.current_hostname}", "")
```

Comment:
```suggestion
        .sub(/\Ahttp(s)?://#{Discourse.current_hostname}\z/, "")
```

I believe this will work too.